.github/workflows/main.yml

```diff
@@ -0,0 +1,197 @@
+        run: .\configure.ps1 -Force
+        shell: powershell
+
+      - name: Build NuGet Packages 📦
+        id: build
+        run: .\build.ps1 -Configuration Release -ReleaseLabel ${{ env.RELEASE_LABEL }} -BuildNumber ${{ env.BUILD_NUMBER }} -Fast -SkipUnitTest -CI
+
+      - name: Run NuGet.Core Unit Tests 🧪
+        run: |
+          Get-ChildItem -Path "test/NuGet.Core.Tests" -Filter "*.csproj" -Recurse | ForEach-Object {
+            dotnet test $_.FullName --configuration Release --no-build --logger trx --results-directory ./build/TestResults/ --verbosity minimal
+          }
+        shell: powershell
+
+      - name: Unit test report
+        uses: dorny/test-reporter@6e6a65b7a0bd2c9197df7d0ae36ac5cee784230c # v2.0.0
+        if: success() || failure()    # run this step even if previous step failed
+        with:
+          name: NuGet Core unit test results
+          path: ./build/TestResults/*.trx
+          reporter: dotnet-trx
+          fail-on-error: true
+
+      - name: Tag release (when release label is final) 🏷️
+        if: ${{ env.RELEASE_LABEL == 'final' }}
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ github.token }}
+          script: |
+            github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: "refs/tags/${{ env.PACKAGE_VERSION }}",
+              sha: context.sha
+            })
+
+      - name: Login to Octopus Deploy 🐙
+        if: env.DEPLOY_TO_OCTOPUS == 'True'
+        uses: OctopusDeploy/login@v1
+        with:
+          server: ${{ secrets.OCTOPUS_URL }}
+          service_account_id: cc09980f-9a73-46c3-8f3f-e3f673740c1f
+
+      - name: Push to Octopus 🐙
+        if: env.DEPLOY_TO_OCTOPUS == 'True'
+        uses: OctopusDeploy/push-package-action@v3
+        with:
+          packages: |
+            ./artifacts/nupkgs/NuGet.CommandLine.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Common.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Configuration.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Frameworks.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Packaging.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Protocol.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Versioning.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.PackageManagement.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Build.Tasks.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Commands.${{ env.PACKAGE_VERSION }}.nupkg
+            ./artifacts/nupkgs/NuGet.Resolver.${{ env.PACKAGE_VERSION }}.nupkg
+
+      - name: Create Release in Octopus 🐙
+        if: env.DEPLOY_TO_OCTOPUS == 'True'
+        uses: OctopusDeploy/create-release-action@v3
+        with:
+          project: Nuget.Client
+          packages: |
```

Comment:
Trivia: If all the packages have the same version you can just specify it once with package_version e.g.

https://github.com/OctopusDeploy/TinyTypes/blob/c99fec7484cc3975bd63dfad0a2caf520c00bfd5/.github/workflows/main.yaml#L105